ci(repo): migrate to github actions

bengaal opened the pull request:
Migrate from travis to github actions.

bengaal committed:
github: add GitHub actions test workflows

* Add tests workflow - this is the tests workflow definition used across
  invenio-app-rdm, invenio-records-resources and invenio-drafts-resources.

* Mark entire tests.yml workflow as raw, to prevent substitution.
  See <https://cookiecutter.readthedocs.io/en/1.7.2/troubleshooting.html#i-m-having-trouble-generating-jinja-templates-from-jinja-templates>.
  We could use _copy_without_render, but that might be unobvious if ever
  someone needs to later add some substitutions.

* Update README to use GitHb Actions workflow badge

bengaal committed:
github(run-tests): change ignore flag

bengaal committed:
github: modify travis references to github

mb-wali commented on invenio_rdm_pure/version.py:
```diff
@@ -11,4 +11,4 @@
 """
 
-__version__ = "1.0.0.dev20200000"
+__version__ = "0.1.0"
```
why change the version - but not a big deal.

bengaal replied:
I just tried to keep it consistent with https://github.com/inveniosoftware/cookiecutter-invenio-module. This was changed [here](https://github.com/inveniosoftware/cookiecutter-invenio-module/commit/e3ca55f18a3e35ed4d3f5875d9647d408c45295b).

mb-wali commented on run-tests.sh:
```diff
@@ -9,7 +9,7 @@
 docker-services-cli up postgresql es redis && \
```
```suggestion

```

mb-wali replied:
```suggestion
docker-services-cli --verbose up es postgresql redis && \
```
using --verbose mode

mb-wali commented on run-tests.sh:
```diff
@@ -9,7 +9,7 @@
 docker-services-cli up postgresql es redis && \
 pydocstyle invenio_rdm_pure tests docs && \
```
```suggestion

```

mb-wali commented on setup.py:
```diff
@@ -109,6 +109,7 @@
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
```
```suggestion

```
we are not sure if this version is yet compatible.

mb-wali commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,40 @@
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+          python-version: [3.6, 3.7, 3.8, 3.9]
```
```suggestion
          python-version: [3.6, 3.7, 3.8]
          requirements-level: [min, pypi]
```

mb-wali reviewed (changes requested):
Request changes:
1. revert all the changes from ```README.rst``` file.
**because**- first we have a PR #26  for this - you can add these changes to that, second you should not remove the coverage badge.
2. remove file ``` invenio_rdm_pure/.github/workflows/pypi-publish.yml```
**because**-  this file has instructions to push the package to PyPI. we don't want that yet. also wrong location.
3. remove file ```invenio_rdm_pure/.github/workflows/tests.yml```
**because** wrong location.
Also see the comments. 

- once changes are done - we will see if something is wrong with the tests failing.

bengaal committed:
github(pypi-publish): remove file

This file has instructions to push the package to PyPI.
We don't want that yet.

bengaal committed:
refactor(setup.py): remove python3.9

Remove Python 3.9 from classifiers list, because we aren't sure
if it's completely compatible.

bengaal committed:
refactor(repo): remove folder invenio_rdm_pure/.github

Files of folder were at wrong location according to
https://github.com/tu-graz-library/invenio-rdm-pure/pull/28.

bengaal committed:
refactor(tests.yml): remove 3.9 from python versions

Remove Python version from list, because we're not sure if it is
compatible.

bengaal committed:
refactor(tests.yml): add requirements-level

Add requirements-level as suggested in
https://github.com/tu-graz-library/invenio-rdm-pure/pull/28.

bengaal committed:
refactor(README.rst): revert changes to readme

Revert changes to README.rst because these changes will be hanled in
another PR: https://github.com/tu-graz-library/invenio-rdm-pure/pull/26.

bengaal committed:
test(run-tests.sh): modify test execution

bengaal committed:
tests(setup.py): add dependency for docker-services-cli

bengaal committed:
tests(setup.py): change sphinx version

mb-wali committed:
ci: generate dependencies

mb-wali commented on setup.py:
```diff
@@ -19,6 +19,8 @@
     "invenio-config>=1.0.3",
     "SQLAlchemy-Utils>=0.33.1,<0.36",  # FIXME without this: ContextualVersionConflict {invenio-files-rest}
+    "docker-services-cli>=0.2.1,<0.3.0",  # TODO: remove once a new release is out
+    "invenio-assets<1.2.0,>=1.1.2",
```
```suggestion
    "invenio-assets>=1.2.0",
```

mb-wali reviewed (commented):
The module is using ```invenio-rdm-records``` of version 0.23.4 - change this to 0.24.1 and remove the ```invenio-assets``` from dependencies.
